socketserver.js

```diff
@@ -52,11 +56,20 @@ module.exports = (io, app) => {
     socket.on('sendMessage', (data) => {
       io.sockets.in(data.room).emit('newMessage', { "message": data.message });
     });
-  });
 
-  // Disconnect the user
-  io.on('disconnect', (socket) => {
-    console.log(`${socket.id} disconnected.`);
-    // console.log(onlineUsers);
+    // Disconnect the user
+    socket.on('disconnect', () => {
+      let index = onlineUsers.indexOf(socket);
+      onlineUsers.splice(index,1);
+      index = rooms.findIndex(x => x.roomID == socket.roomID);
+      if(rooms[index].isFilled == true){
+        rooms[index].isFilled = false;
```

Comment:
delete the whole room here.

Reply:
since one user will be removed, the other will be in the joined in the same room here. when another user comes, he ll be getting joined with the existing user without the existing user's consent. thus we need to dlete the room and create a new one when the existing user wants to join agian.